staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest_test.go

```diff
@@ -922,7 +927,10 @@ func TestFinishRequest(t *testing.T) {
 	}
 	for i, tc := range testcases {
 		t.Run(tc.name, func(t *testing.T) {
+			ctx, cancel := tc.timeout()
```

Comment:
will the WithSecondaryDeadline timeout trigger this one too? I.e. is there only one timeout value in a context?

Reply:
context is hierarchical, if a parent context times out or is canceled then the child context also gets canceled/times out.
```
	parent, c1 := context.WithTimeout(context.TODO(), 5 * time.Second)
	child, c2 := context.WithTimeout(parent, 10 * time.Second)
```
Here the child context will also time out in `5s`.